test/test_sparse_semi_structured.py

```diff
@@ -596,16 +597,16 @@ def setUp(self):
         else:
             SparseSemiStructuredTensor._FORCE_CUTLASS = False
 
-
+    @parametrize("out_dtype", CUSPARSELT_MIXED_DTYPE_SUPPORT)
     @parametrize("dense_input_shape", [(128, 128)])
-    def test_cslt_sparse_mm_int8_in_fp16_out(self, dense_input_shape, device):
+    def test_cslt_sparse_mm_mixed_dtype(self, dense_input_shape, out_dtype, device):
         A = rand_sparse_semi_structured_mask(128, 128, dtype=torch.int8)
         A_compressed = torch._cslt_compress(A)
 
         B = torch.rand(dense_input_shape, device=device).to(torch.int8)
 
-        dense_result = torch.mm(A.cpu().to(torch.int64), B.t().cpu().to(torch.int64)).to(device, dtype=torch.float16)
-        sparse_result = torch._cslt_sparse_mm(A_compressed, B.t(), out_dtype=torch.float16)
+        dense_result = torch.mm(A.cpu().to(torch.int64), B.t().cpu().to(torch.int64)).to(device, dtype=out_dtype)
+        sparse_result = torch._cslt_sparse_mm(A_compressed, B.t(), out_dtype=out_dtype)
```

Comment:
Just in case (you're 99% likely already aware): I think this is fine since this is a private op, but using the higher order op out_dtype is what we'll want people to use to interact with this.